packages/scandipwa/src/component/CategoryConfigurableAttributes/CategoryConfigurableAttributes.component.js

```diff
@@ -119,6 +128,15 @@ export class CategoryConfigurableAttributes extends ProductConfigurableAttribute
         switch (attribute_code) {
         case 'price':
             return this.renderPriceSwatch(option);
+        case 'category_id':
+            const childrenCategoryIds = this.props.childrenCategories.map((category) => category.id.toString());
+            const subCategoriesIds = option.attribute_values.filter((item) => childrenCategoryIds.includes(item));
+            if (subCategoriesIds.length) {
+                this.props.configurable_options.category_id.attribute_values = subCategoriesIds;
```

Comment:
Please prefer destructuring assignment over the using long chains of properties.